### ISSUE TITLE
Easier support for monorepos

### ISSUE THREAD

**Issue by maneetgoyal** [christopherthielen/typedoc-plugin-external-module-name#423]
Hi All:

**Problem**:
- Currently, this plugin requires declaring `@module` on each of the `.ts` file that we want to be merged together.
- In a monorepo setup, we can have nested repositories. And each of those repositories will have their own `package.json`.
- A monorepo user would like to have the symbols (functions, enums, interfaces, etc.) clubbed together based on the child repository they are in, but adding a `@module` declaration in each of the numerous possible files is very tedious.

---

**Proposed Solution**:
- It will be very useful if an **option** can be provided in `typedoc.json` via this plugin which specifies the relative path of all those "children" repositories and then the plugin can merge all the symbols appropriately based on child repository name.
- A [fork](https://github.com/asgerjensen/typedoc-plugin-external-module-map) of this plugin does it well but doesn't seem to have active maintenance schedule. `typedoc-plugin-external-module-name` is probably the most up-to-date and active plugin amongst all its forks in terms of TypeDoc version compatibility.
- Will be super helpful if this feature can be provided. This plugin seems to already have all the tools needed for this functionality.

---

**Comment by christopherthielen** [comment]
Agreed, this would be great. I am also sick of putting `@module` everywhere in my source code.
I see what the fork is doing using regular expressions, and that seems OK.  

I don't understand your proposal though.  Can you clarify what you meant by:

> It will be very useful if an option can be provided in typedoc.json via this plugin which specifies the relative path of all those "children" repositories and then the plugin can merge all the symbols appropriately based on child repository name.

**Comment by maneetgoyal** [comment]
@christopherthielen The fork we are taking about, accepts a path of the concerned modules in a "single" repository and categorizes or clubs the symbols/interfaces present inside those modules based on the module name.  

Whereas in a monorepo, there are multiple child repositories. So if I provide a regex to the path of those child repositories, then all the types/interafces/symbols should be clubbed together based on the child repositiories' names without us requiring the need for adding `@module`. The TypeDoc will look something like:

<img width="408" alt="Screen Shot 2020-04-29 at 12 18 01 AM" src="https://user-images.githubusercontent.com/25290212/80570148-26b68780-89af-11ea-9316-b89056a16b07.png">

So what I am describing is basically the same thing which that fork is doing but instead of modules, I have child repositories.

At the time I raised this issue, the fork was a bit outdated but now they have upgraded their TS too. I have also switched to [unstubbable/typedoc-plugin-monorepo](https://github.com/unstubbable/typedoc-plugin-monorepo) which is doing what I described above. So my requirement is currently met but it will a great enhancement for this plugin nonetheless.
